RestaurantReservation.Db/Repositories/CustomerRepository.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Tokens;
+using RestaurantReservation.Db.Models;
+
+namespace RestaurantReservation.Db.Repositories;
+public class CustomerRepository
+{
+    private RestaurantReservationDbContext _dbContext;
+    public CustomerRepository(RestaurantReservationDbContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
+
+    public async Task<Customer?> AddCustomerAsync(Customer customer)
+    {
+        if (customer == null)
```

Comment:
The method is not expecting a nullable customer, so it won't be null, either remove it or make the customer parameter nullable